### PR TITLE
Add Location Dialog

### LINKED ISSUE #72: View: Location Dialog
# Description

As user, I want to have a dialog to add and edit locations.

# Acceptance criteria

- User can open the dialog in `Program` and `Add Event` pages
- User can enter names for new locations
- User can add locations to the list
- User can not add a location with a name that is already in the list
- User can remove locations from the list
- User can save the changes
- User can close the dialog without saving

Should look like this:

image

# Possible way to test

- Click the `Locations` edit button in `Program` and `Add Event`
- Enter a location name, click add
- Click add again to see if you can add the same location a second time
- Click the delete button next to a location
- Click the save button
- Open the dialog again and see if the changes were saved
- Click the cancel button

### PR DESCRIPTION
**Pull request by bognern**
# 💻 What

Added location dialog with basic add/remove functionality.
Filled location and category dialogs with test data.
Minor UI adjustments.

# 📘 How

Reuses UI from #21 Category Dialog.
Pass common component `event-dialog` separate lists for the location and category dialogs to populate them accordingly.
Checks if location/category name already exists before adding it.

# 👀 See

in `/program/add`
image

in `/program/edit/1`
image

image

### CODE REVIEW

File: src/app/program/shared/event-dialog/event-dialog.component.ts [hunk 1]
```diff
@@ -1,31 +1,42 @@
...
+    const newItemName: string | null =
+      this.itemFormGroup.controls['name'].value;
 
-    if (this.newItem && !this.itemList.includes(this.newItem)) {
-      this.itemList.push(this.newItem);
-      this.newItem = '';
+    if (newItemName && this.itemList) {
+      const exists = this.itemList.some((item) => item.name === newItemName);
+      if (!exists) {
+        const newItem: Item = { id: 0, name: newItemName };
```
Comment (marioradman) [inline src/app/program/shared/event-dialog/event-dialog.component.ts:37]: why are we setting the id hardcoded to zero?

**Review by marioradman — COMMENTED** [review]
LGTM

File: src/app/program/shared/event-dialog/event-dialog.component.ts (hunk 1 above)
Comment (bognern) [reply src/app/program/shared/event-dialog/event-dialog.component.ts:37]: I'm aware that this is not an ideal solution. To add an item to the list defined as `Item[]{id: number, name: string}`, I need to pass it an ID. As soon as I find out what the request to update the list is going to look like, I will adjust the code accordingly.

**Review by Islam-Hemida — APPROVED** [review]
all fine

**Pull request merged**